pkg/validate/serialize.go

```diff
@@ -31,14 +54,93 @@ func Verify(yamlFileName string) error {
 // (1) the yaml file can not be converted to json.
 // (2) there is a problem while unmarshalling in go type.
 // Returns an object of type olm.ClusterServiceVersion.
-func unmarshal(rawYAML []byte) (*olm.ClusterServiceVersion, error) {
+func unmarshal(rawYAML []byte) (olm.ClusterServiceVersion, error) {
 
 	var csv olm.ClusterServiceVersion
 
-	if err := yaml.Unmarshal(rawYAML, &csv); err != nil {
-		return nil, fmt.Errorf("error parsing CSV list (JSON) : %s", err)
+	rawJson, err := yaml.YAMLToJSON(rawYAML)
+	if err != nil {
+		fmt.Printf("error parsing raw YAML to Json: %s", err)
+		return csv, err
+	}
+	if err := json.Unmarshal(rawJson, &csv); err != nil {
+		return csv, fmt.Errorf("error parsing CSV list (JSON) : %s", err)
+	}
+
+	return csv, nil
+}
+
+func hasEmptyField(val interface{}) []string {
+
+	fieldValue := reflect.ValueOf(val)
+
+	switch fieldValue.Kind() {
+	case reflect.Struct:
+		return checkStructHasEmptyFields(fieldValue, []string{})
+	default:
+		return []string{"Error: input file is not aligned with Operator-Lifecycle-Manager's CSV type."}
 	}
+}
+
+// Takes in a string slice and checks if a string (x) is present in the slice.
+// Return true if the string is present in the slicec.
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if strings.ToLower(x) == strings.ToLower(n) {
```

Comment:
I want it to ignore case while comparing the strings.

Reply:
I could also use `strings.EqualFold` for that if you think that's more appropriate.

Reply:
I understand but is there a situation where `omitempty` has upper case letters?

Reply:
Not for `omitempty`, but I was thinking of using this helper in some other part of the code. Although, that did not work out, I decided to make it more general so it can be used in other parts.